### PR TITLE
Add NixOS support

### PR DESCRIPTION
**Pull request by midchildan**
This PR is part of an attempt to package Traffic Server in [Nixpkgs](https://github.com/NixOS/nixpkgs).

### CODE REVIEW

File: config.layout [hunk 1]
```diff
@@ -277,3 +277,23 @@
...
+    sysconfdir:    /etc+
+    datadir:       ${prefix}/share+
+    docdir:        ${prefix}/share/doc+
+    installbuilddir: ${datadir}/build
+    includedir:    ${prefix}/include
+    localstatedir: /var/lib+
+    runtimedir:    /run+
+    logdir:        /var/log+
+    cachedir:      /var/cache+
+</Layout>
```
Comment (midchildan) [inline config.layout:299]: About half of these are replaced through configure flags when building with Nix, but I still added them for the sake of consistency.

File: tools/tsxs.in [hunk 2]
```diff
@@ -204,6 +204,9 @@ do
 		OBJ=$OPTARG
 		;;
 	i)
+		if [[ "$INSTALLDIR" = /nix/store* ]]; then
+			bail "unsupported option: -i is disabled on Nix"
+		fi
```
Comment (midchildan) [inline tools/tsxs.in:209]: `/nix/store` is where packages are installed and is non-writable.

File: rc/trafficserver.service.in [hunk 3]
```diff
@@ -17,7 +17,7 @@
 #
 [Unit]
 Description=Apache Traffic Server is a fast, scalable and extensible caching proxy server.
-After=syslog.target network.target
+After=network.target
```
Comment (midchildan) [inline rc/trafficserver.service.in:20]: `syslog.target` is [removed from recent versions of systemd](https://github.com/systemd/systemd/blob/1cee1c52833fb6e3829e510109404852a17e5bdd/NEWS#L8931).

File: rc/trafficserver.service.in [hunk 4]
```diff
@@ -26,10 +26,7 @@ ExecStart=@exp_bindir@/traffic_manager $TM_DAEMON_ARGS
 Restart=on-failure
 RestartSec=5s
 LimitNOFILE=1000000
-ExecStopPost=/bin/sh -c ' \
-        export TM_PIDFILE=$(@exp_bindir@/traffic_layout 2>/dev/null | grep RUNTIMEDIR | cut -d: -f2)/manager.lock ; \
-        /bin/rm $TM_PIDFILE ; \
-        if [[ $? -ne 0 ]]; then echo "ERROR: Unable to delete PID"; exit 1; fi'
+PIDFile=@exp_runtimedir@/manager.lock
```
Comment (midchildan) [inline rc/trafficserver.service.in:29]: Using `PIDFile` to clean up PID files is slightly more portable, as some distros don't have `/bin/rm`.

**Comment by shinrich** [comment]
[approve ci]

**Comment by randall** [comment]
[approve ci]

**Comment by randall** [comment]
[approve ci]
[approve ci]

**Comment by randall** [comment]
[approve ci rat]

**Review by bryancall — APPROVED** [review]

**Pull request merged**